Objects/tupleobject.c

```diff
@@ -53,7 +54,11 @@ tuple_alloc(Py_ssize_t size)
                 sizeof(PyObject *))) / sizeof(PyObject *)) {
         return (PyTupleObject *)PyErr_NoMemory();
     }
-    return PyObject_GC_NewVar(PyTupleObject, &PyTuple_Type, size);
+    PyTupleObject *result = PyObject_GC_NewVar(PyTupleObject, &PyTuple_Type, size);
+    if (result != NULL) {
+        _PyTuple_RESET_HASH_CACHE(result);
```

Comment:
Maybe `_PyTuple_RESET_HASH_CACHE` could handle the NULL values and by-pass them?

Reply:
Looking at this, there are more places where non-NULL is already known than not, so this is just an extra unnecessary check.  The compiler can probably optimize that out, but I think an `assert(op != NULL)` would be better than `if (op != NULL)`.